ckanext/security/controllers.py

```diff
@@ -129,7 +129,13 @@ def login(self):
             res['mfaConfigured'] = mfaConfigured
 
             if identity['mfa']:
-                res['mfaCodeValid'] = totp_challenger.check_code(identity['mfa'], verify_only=True)
+                code_valid = totp_challenger.check_code(identity['mfa'], verify_only=True)
+                res['mfaCodeValid'] = code_valid and not locked_out
+                if code_valid:
+                    log.info('Login succeeded for {}'.format(user_name))
```

Comment:
Should use parameterised logging instead of direct string formatting. Eg `log.info('Login succeeded for %s', user_name)`

Reply:
Right, thanks... I wasn't aware of the logger having the ability to format in this way. Looks like that is the preferred approach to prevent interpolation occurring regardless of the logging level.